packages/featureflags/featureToggleManager.js

```diff
@@ -7,89 +7,101 @@ const { getRedisClient } = require('@open-condo/keystone/redis')
+
     constructor () {
-        try {
-            const config = JSON.parse(FEATURE_TOGGLE_CONFIG)
-            featureToggleApiUrl = config.url
-            featureToggleApiKey = config.apiKey
-        } catch (e) {
-            logger.error({ msg: 'parse FEATURE_TOGGLE_CONFIG error', error: e })
+        if (FEATURE_TOGGLE_CONFIG.url && FEATURE_TOGGLE_CONFIG.apiKey) {
+            this._url = `${FEATURE_TOGGLE_CONFIG.url}/${FEATURE_TOGGLE_CONFIG.apiKey}`
+            this._static = null
+        } else if (FEATURE_TOGGLE_CONFIG.static) {
+            // NOTE(pahaz): value example: {"sms-after-ticket-creation":{"defaultValue":false,"rules":[{"condition":{"organization":{"$in":[]}},"force":true}]},"refetch-tickets-in-control-room":{"defaultValue":false,"rules":[{"force":true}]},"ticket-import":{"defaultValue":false,"rules":[{"condition":{"isSupport":true},"force":true}]},"send-billing-receipts-notifications-task":{"defaultValue":true},"max-count-completed-ticket-to-close-for-organization-task":{"defaultValue":100}}
+            this._url = null
+            this._static = FEATURE_TOGGLE_CONFIG.static
+        } else {
+            this._url = null
+            this._static = {}
+            logger.warn('No FEATURE_TOGGLE_CONFIG! Every features and values will be false!')
         }
+        this._redisKey = REDIS_FEATURES_KEY
+        this._redisExpires = FEATURES_EXPIRED_IN_SECONDS
     }
 
     async fetchFeatures () {
-        try {
-            const redisClient = getRedisClient()
-            const cachedFeatureFlags = await redisClient.get(REDIS_FEATURES_KEY)
-
-            if (cachedFeatureFlags) return JSON.parse(cachedFeatureFlags)
+        if (this._url) {
+            try {
+                const cachedFeatureFlags = await this.redis.get(this._redisKey)
+                if (cachedFeatureFlags) return JSON.parse(cachedFeatureFlags)
 
-            if (featureToggleApiUrl && featureToggleApiKey) {
-                const fetchedFeatureFlags = await fetch(`${featureToggleApiUrl}/${featureToggleApiKey}`)
+                const fetchedFeatureFlags = await fetch(this._url)
                     .then((res) => res.json())
                     .then((parsed) => {
                         return Promise.resolve(parsed.features)
                     })
 
-                redisClient.set(REDIS_FEATURES_KEY, JSON.stringify(fetchedFeatureFlags), 'EX', FEATURES_EXPIRED_IN_SECONDS)
-
+                await this.redis.set(this._redisKey, JSON.stringify(fetchedFeatureFlags), 'EX', this._redisExpires)
                 return fetchedFeatureFlags
-            } else {
-                throw new Error(WRONG_FEATURE_TOGGLE_CONFIG_ERROR)
+            } catch (err) {
+                logger.error({ msg: 'fetchFeatures error', err })
             }
-        } catch (e) {
-            logger.error({ msg: 'fetchFeatures error', error: e })
-
-            return {}
+        } else if (this._static) {
+            return JSON.parse(JSON.stringify(this._static))
```

Comment:
Why do you need deep copy here? Aren't it static?

Reply:
To avoid global settings modifiactions by developer mistake or typo.